src/nets/e2e_asr_attctc_th.py

```diff
@@ -288,18 +307,18 @@ def forward(self, data):
         # utt list of olen
         ys = [np.fromiter(map(int, tids[i]), dtype=np.int64)
               for i in sorted_index]
-        if self.training:
-            ys = [to_cuda(self, Variable(torch.from_numpy(y))) for y in ys]
+        if torch_is_old:
+            ys = [to_cuda(self, Variable(torch.from_numpy(y), volatile=not self.training)) for y in ys]
         else:
-            ys = [to_cuda(self, Variable(torch.from_numpy(y), volatile=True)) for y in ys]
+            ys = [to_cuda(self, torch.from_numpy(y)) for y in ys]
 
         # subsample frame
         xs = [xx[::self.subsample[0], :] for xx in xs]
         ilens = np.fromiter((xx.shape[0] for xx in xs), dtype=np.int64)
-        if self.training:
-            hs = [to_cuda(self, Variable(torch.from_numpy(xx))) for xx in xs]
-        else:
+        if torch_is_old and not self.training:
             hs = [to_cuda(self, Variable(torch.from_numpy(xx), volatile=True)) for xx in xs]
```

Comment:
It is better to use same expression in the case of `ys`.
```
if torch_is_old:
              hs = [to_cuda(self, Variable(torch.from_numpy(xx), volatile=not self.training)) for xx in xs]
```

Reply:
gotcha